src/autogluon/cloud/backend/backend.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional, Union
+
+import pandas as pd
+
+from ..endpoint.endpoint import Endpoint
+
+
+class Backend(ABC):
```

Comment:
This might make sense, but requires some thinking and design. Let's keep it for now in this PR and do it in other PRs if we think it's worth it. This PR is more about extracting the backend